### DISCUSSION TITLE
is there a way of controlling multiple checkboxes with one and vice versa?

### DISCUSSION THREAD

**Discussion by goongoguma** [react-hook-form/react-hook-form#8563] [Q&A]
Hi

I am trying to make group of checkboxes that controlled by one checkbox and vice versa.

Here is the issue that I reproduced in [here](https://codesandbox.io/s/sweet-antonelli-qcu3j3).

In the example, when **All Vegetables** is checked, the rest of checkboxes are checked.

On the other hand, If **All Vegetables** is unchecked, the rest of them are unchecked as well.

Here is the part I am stuck.

When one of options is unchecked,  **All Vegetables** is also supposed to be unchecked but it does not.

And when all the options is checked, **All Vegetables** should also be checked but it does not.

How can I make group of checkboxes interactive?

(PS. I also used dirtyfield property in formstate, however in real project, the checkbox group is in a modal. Therefore dirtyfield is reset whenever the modal is closed.)

**Comment by bluebill1049** [comment]
I would suggest building a controlled checkbox group and then hook it up with `useController`, it's one of the benefits of building a controlled component, your component state could easily represent your data modal, and `controller` or `useController` will simply by the data supplier and change subscriber.
